### PR TITLE
Bugfix/223/popup window is showing twice on the loginsignup form

### LINKED ISSUE #223: Popup window is showing twice on the Login\Signup form
When we on the Login/Signup form trying to close the window, we see 2 modal popup-windows instead of 1

Image

Image

### PR DESCRIPTION
**Pull request by holus-bolus**
Created the additional rules for the dialog window. This window is used in several components, so I had to change the logic for the hook, because we need to show different messages for different states.
modal-context
signup window
signup
stepper window

### CODE REVIEW

**Review by VictoriaRodik — APPROVED** [review]

**Review by iva-stasia — APPROVED** [review]

**Review by iaremchuk — APPROVED** [review]

**Review by OliaHrytsak — APPROVED** [review]

File: src/components/user-steps-wrapper/UserStepsWrapper.jsx [hunk 1]
```diff
@@ -56,7 +56,12 @@ const UserStepsWrapper = ({ userRole }) => {
           <StepProvider initialValues={initialValues} stepLabels={stepLabels}>
             <StepWrapper steps={stepLabels}>{childrenArr}</StepWrapper>
           </StepProvider>
-        )
+        ),
+        dialogConfig: {
+          title: 'Please Confirm',
+          message:
+            'Are you certain you want to close? Any unsaved changes will be lost'
```
Comment (vlad-khrychov) [inline src/components/user-steps-wrapper/UserStepsWrapper.jsx]: Please move this string in `.json` file in `constants` folder.

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

File: src/components/user-steps-wrapper/UserStepsWrapper.jsx (hunk 1 above)
Comment (holus-bolus) [reply src/components/user-steps-wrapper/UserStepsWrapper.jsx]: @vlad-khrychov changed messages to the predefined ones

File: src/components/user-steps-wrapper/UserStepsWrapper.jsx [hunk 2]
```diff
@@ -56,7 +57,11 @@ const UserStepsWrapper = ({ userRole }) => {
           <StepProvider initialValues={initialValues} stepLabels={stepLabels}>
             <StepWrapper steps={stepLabels}>{childrenArr}</StepWrapper>
           </StepProvider>
-        )
+        ),
+        dialogConfig: {
+          title: titles.confirmTitle,
+          message: questions.unsavedChanges
```
Comment (vlad-khrychov) [inline src/components/user-steps-wrapper/UserStepsWrapper.jsx]: Add `const { t } = useTranslation()` in components and context.
```suggestion
          title: t(item.titles.confirmTitle),
          message: t(questions.unsavedChanges)
```

File: src/context/modal-context.jsx [hunk 3]
```diff
@@ -6,32 +7,46 @@ import {
...
 
 const ModalContext = createContext({})
 
 const ModalProvider = ({ children }) => {
   const [modal, setModal] = useState(null)
   const [paperProps, setPaperProps] = useState({})
   const [confirmDialogOpen, setConfirmDialogOpen] = useState(false)
+  const [dialogConfig, setDialogConfig] = useState({
+    title: titles.confirmTitle,
+    message: questions.unsavedChanges
```
Comment (vlad-khrychov) [inline src/context/modal-context.jsx]:
```suggestion
    title: t(titles.confirmTitle),
    message: t(questions.unsavedChanges)
```

File: src/context/modal-context.jsx [hunk 4]
```diff
@@ -57,11 +72,11 @@ const ModalProvider = ({ children }) => {
         <ConfirmDialog
           cancelButton='No'
           confirmButton='Yes'
-          message='Are you sure you want to close?'
+          message={dialogConfig.message}
```
Comment (vlad-khrychov) [inline src/context/modal-context.jsx]:
```suggestion
          message={t(dialogConfig.message)}
```

File: src/context/modal-context.jsx [hunk 5]
```diff
@@ -57,11 +72,11 @@ const ModalProvider = ({ children }) => {
...
           onConfirm={handleConfirmClose}
           onDismiss={() => setConfirmDialogOpen(false)}
           open={confirmDialogOpen}
-          title='Confirm Close'
+          title={dialogConfig.title}
```
Comment (vlad-khrychov) [inline src/context/modal-context.jsx]:
```suggestion
          title={t(dialogConfig.title)}
```

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

File: src/context/modal-context.jsx (hunk 5 above)
Comment (holus-bolus) [reply src/context/modal-context.jsx]: Thank you for pointing on my mistakes, I changed the messages and translations

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**